File: bubblesort.c
/* Bubble Sort */
#include<stdio.h>
void bubblesort(int* a, int len){
	int flag = 0;
	int i,j,temp;
	for(i=1; i<len; i++){
		flag = 0;
		for(j=1; j<len-i+1; j++){
			if(a[j-1] > a[j]){
				flag++;
				temp = a[j-1];
				a[j-1] = a[j];
				a[j] = temp;
			}
		}
		if (flag == 0) break;
	}
}
int main(int argc, char** argv){
	int i;
	int arr[] = {5,2,3,4};
	int length = sizeof(arr)/sizeof(arr[0]);
	bubblesort(arr,length);
	printf("Sorted Array: ");
	for(i=0; i<length; i++){
		printf("%d ", arr[i]);
	}
	printf("\n");
	return 0;
}

File: bubblesort.py
def bubbleSort(li, length):
	flag = 0
	for i in range(1, length):
		for j in range(1, length - i + 1):
			if li[j - 1] > li[j]:
				flag = flag + 1
				li[j-1], li[j] = li[j], li[j-1]
		if flag == 0:
			break
def main():
	li = [5, 2, 3, 4]
	length = len(li)
	print("Unsorted List: ")
	print(li)

	bubbleSort(li, length)

	print("Sorted List: ");
	print(li)

if __name__ == "__main__":
	main()
